src/Microsoft.DocAsCode.MarkdownLite/JsonTokenTreeRenderer.cs

```diff
@@ -219,58 +224,68 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken
+            return Insert(token, $"Ignore>{Escape(token.SourceInfo.Markdown)}");
         }
 
         public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
         {
-            return this.Insert($"Raw(From{token.Rule.Name})", GetSize(), Type.Leaf);
+            return Insert(token, $"Raw(From{token.Rule.Name})>{Escape(token.SourceInfo.Markdown)}");
         }
 
         #endregion
 
         #region Private
 
-        private enum Type
+        protected StringBuffer Insert(IMarkdownToken token, StringBuffer name, StringBuffer tokenContent = null)
         {
-            Leaf,
-            NonLeaf
+            // TODO: separate the name to extra properties
+            int startLineNumber = token.SourceInfo.LineNumber;
+            int endLineNumber = (token.SourceInfo.ValidLineCount > 0) ? (startLineNumber + token.SourceInfo.ValidLineCount - 1) : startLineNumber;
+            StringBuffer result = $"{{\"name\":\"{startLineNumber}>{endLineNumber}>{name}\"";
+            if (tokenContent != null)
+            {
+                // If tokenContent is not empty ,should remove the last character(',')
+                if (tokenContent.GetLength() > 0 && tokenContent.EndsWith(','))
+                {
+                    // TODO: add a method 'remove' of StringBuffer
+                    string contentTemp = tokenContent;
+                    tokenContent = contentTemp.Remove(contentTemp.Length - 1);
+                }
+                result += $",\"children\":[{tokenContent}]";
+            }
+            result += "},";
+
+            return result;
         }
 
-        private StringBuffer Insert(StringBuffer name, StringBuffer tokenContent, Type tokenType)
+        protected StringBuffer Insert(int startLineNumber, int endLineNumber, StringBuffer name, StringBuffer tokenContent = null)
         {
-            StringBuffer result = $"{{\"name\":\"{name}\"";
-            switch (tokenType)
+            StringBuffer result = $"{{\"name\":\"{startLineNumber}>{endLineNumber}>{name}\"";
+            if (tokenContent != null)
             {
-                case Type.NonLeaf:
-                {
-                    // If tokenContent is not empty ,should remove the last character(',')
-                    if (tokenContent.GetLength() > 0 && tokenContent.EndsWith(','))
-                    {
-                        // TODO: add a method 'remove' of StringBuffer
-                        string contentTemp = tokenContent;
-                        tokenContent = contentTemp.Remove(contentTemp.Length - 1);
-                    }
-                    result += $",\"children\":[{tokenContent}]}},";
-                    break;
-                }
-                case Type.Leaf:
+                // If tokenContent is not empty ,should remove the last character(',')
+                if (tokenContent != StringBuffer.Empty && tokenContent.EndsWith(','))
```

Comment:
>tokenContent != StringBuffer.Empty [](start = 20, length = 34)

can remove the check? EndsWith would check the length and return false, and line 248